chain/client/src/client.rs

```diff
@@ -1798,11 +1832,134 @@ impl Client {
+            block_status.in_orphan_for = now.checked_duration_since(*added);
+            block_status.chunk_hashes =
+                block.chunks().iter().map(|it| it.chunk_hash()).collect_vec();
+        });
+
+        // Fetch the status of the chunks.
+        for height_entry in height_status_map.iter_mut() {
+            for block_entry in height_entry.1.iter_mut() {
+                for chunk_hash in block_entry.1.chunk_hashes.iter() {
+                    match self.chain.chain_store().chunk_exists(&chunk_hash) {
+                        Ok(true) => block_entry.1.chunks_completed.insert(chunk_hash.clone()),
+                        _ => block_entry.1.chunks_not_completed.insert(chunk_hash.clone()),
+                    };
+                }
+            }
+        }
+        let use_colour = matches!(self.config.log_summary_style, LogSummaryStyle::Colored);
+        let paint = |colour: ansi_term::Colour, text: Option<String>| match text {
+            None => ansi_term::Style::default().paint(""),
+            Some(text) if use_colour => colour.bold().paint(text),
+            Some(text) => ansi_term::Style::default().paint(text),
+        };
+
+        // Returns a status line for each block - also prints what is happening to its chunks.
+        let next_blocks_log = height_status_map
+            .keys()
+            .sorted()
+            .map(|height| {
+                let val = height_status_map.get(height).unwrap();
+
+                let block_debug = val
+                    .iter()
+                    .map(|entry| {
+                        let block_info = entry.1;
+                        let chunk_status = block_info
+                            .chunk_hashes
+                            .iter()
+                            .map(|it| {
+                                if block_info.chunks_completed.contains(it) {
+                                    "✔"
+                                } else if block_info.chunks_received.contains(it) {
+                                    "⬇"
+                                } else if block_info.chunks_requested.contains(it) {
+                                    "⬆"
+                                } else {
+                                    "."
+                                }
+                            })
+                            .collect::<Vec<&str>>();
+
+                        let chunk_status_color =
+                            if block_info.chunks_completed.len() == block_info.chunk_hashes.len() {
+                                paint(ansi_term::Colour::Green, Some(chunk_status.join("")))
+                            } else {
+                                paint(ansi_term::Colour::White, Some(chunk_status.join("")))
+                            };
+
+                        format!(
+                            "{} {:?} {:?} Chunks:({}))",
+                            entry.0,
+                            block_info.in_progress_for,
```

Comment:
Could you add "in progress for" and "in orphan for" before the timestamps? Otherwise it's hard to know what the time stands for.

Reply:
done